JAICore/jaicore-ml/src/main/java/ai/libs/jaicore/ml/scikitwrapper/ScikitLearnClassificationWrapper.java

```diff
@@ -0,0 +1,45 @@
+package ai.libs.jaicore.ml.scikitwrapper;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.api4.java.ai.ml.core.dataset.schema.attribute.ICategoricalAttribute;
+import org.api4.java.ai.ml.core.dataset.supervised.ILabeledDataset;
+import org.api4.java.ai.ml.core.dataset.supervised.ILabeledInstance;
+import org.api4.java.ai.ml.core.evaluation.IPrediction;
+import org.api4.java.ai.ml.core.evaluation.IPredictionBatch;
+import org.api4.java.ai.ml.core.exception.PredictionException;
+import org.api4.java.ai.ml.core.exception.TrainingException;
+
+import ai.libs.jaicore.ml.classification.singlelabel.SingleLabelClassification;
+import ai.libs.jaicore.ml.classification.singlelabel.SingleLabelClassificationPredictionBatch;
+import ai.libs.jaicore.ml.core.EScikitLearnProblemType;
+
+public class ScikitLearnClassificationWrapper<P extends IPrediction, B extends IPredictionBatch> extends AScikitLearnWrapper<P, B> {
+
+	public ScikitLearnClassificationWrapper(final String pipeline, final String imports) throws IOException {
+		super(EScikitLearnProblemType.CLASSIFICATION, pipeline, imports);
+	}
+
+	@Override
+	protected boolean doLabelFitToProblemType(final ILabeledDataset<? extends ILabeledInstance> data) {
+		return data.getLabelAttribute() instanceof ICategoricalAttribute;
+	}
+
+	@SuppressWarnings("unchecked")
+	@Override
+	protected B handleOutput(final File outputFile) throws PredictionException, TrainingException {
+		List<List<Double>> rawLastPredictionResults = this.getRawPredictionResults(outputFile);
+
+		if (rawLastPredictionResults.get(0).size() == 1) {
```

Comment:
Might be good, if "rawLastPredictionResults" is really not empty before accessing the 0th element.